packages/attune/src/bin/attune/cmd/apt/pkg/remove.rs

```diff
@@ -0,0 +1,176 @@
+use std::{iter::once, process::ExitCode};
+
+use axum::http::StatusCode;
+use clap::Args;
+use gpgme::{Context, ExportMode, Protocol};
+use percent_encoding::percent_encode;
+use tracing::debug;
+
+use crate::config::Config;
+use attune::{
+    api::{ErrorResponse, PATH_SEGMENT_PERCENT_ENCODE_SET},
+    server::repo::index::{
+        PackageChange, PackageChangeAction,
+        generate::{GenerateIndexRequest, GenerateIndexResponse},
+        sign::{SignIndexRequest, SignIndexResponse},
+    },
+};
+
+#[derive(Args, Debug)]
+pub struct PkgRemoveCommand {
+    /// Name of the repository to remove the package from
+    #[arg(long, short)]
+    repo: String,
+    /// Distribution to remove the package from
+    #[arg(long, short)]
+    distribution: String,
+    /// Component to remove the package from
+    #[arg(long, short)]
+    component: String,
+
+    /// GPG key ID to sign the index with (see `gpg --list-secret-keys`)
+    #[arg(long, short)]
+    key_id: String,
+
+    /// Name of the package to remove
+    #[arg(long, short)]
+    package: String,
+    /// Version of the package to remove
+    #[arg(long, short)]
+    version: String,
+    /// Architecture of the package to remove
+    #[arg(long, short)]
+    architecture: String,
+}
+
+pub async fn run(ctx: Config, command: PkgRemoveCommand) -> ExitCode {
+    // Remove the package from the index, retrying if needed.
```

Comment:
[nit] I don't actually see any retries here- although if we want to do some I recommend we use `backon`, I can throw together a convenient extension to reqwest types to use it.